### DISCUSSION TITLE
Why isn't KLV-metadata stream handled/received by RTSP?

### DISCUSSION THREAD

**Discussion by Hippakaveri** [bluenviron/mediamtx#2690] [Questions]
### Question

I've got a stream as bellow : 
"Input #0, mpegts, from 'udp://239.192.2.1:50086':
  Duration: N/A, start: 5898.369000, bitrate: N/A
  Program 1
    Metadata:
      service_name    : Service01
      service_provider: FFmpeg
  Stream #0:0[0x100]: Video: h264 (High) ([27][0][0][0] / 0x001B), yuv420p(progressive), 1280x722, 90k tbr, 90k tbn
  Stream #0:1[0xfc]: Data: klv (KLVA / 0x41564C4B)" 
 And if I configure RTSP as such
 "  udpSource:
    source: udp://239.192.2.1:50086 " 

Why does it only handle the video stream and not the klv as well? 

If I ffprobe rtsp://127.0.0.1:8554/
I only get :
"Input #0, rtsp, from 'rtsp://127.0.0.1:8554/udpSource':
  Metadata:
    title           :
  Duration: N/A, start: 0.155011, bitrate: N/A
  Stream #0:0: Video: h264 (High), yuv420p(progressive), 1280x722, 30 tbr, 90k tbn"

Is this a bug or working as intended? 

Thanks.

**Comment by aler9** [comment]
Hello, the reason is that the server currently routes tracks that are compatible with RTSP, and KLV metadata are not.

Nothing prevents from improving the server in order to add KLV support to RTSP, or alternatively, allow routing tracks that are not RTSP-compatible. I've created a feature request here: #2693

**Comment by Hippakaveri** [comment]
Seems like I can forward the streams from udp to rtsp using following gstreamer pipe "gst-launch-1.0 udpsrc address=<uni/multicast address> reuse=true port=<port> ! tsparse ! queue ! rstspclientsink location=rtsp://localhost:$RTSP_PORT/$MTX_PATH" . Which is a lot simpler than I thought and had tried in the past. Hope this is of use for someone.

**Reply by aler9** [reply]
What you wrote works indeed, the problem is that it creates a single MPEG-TS inside RTSP, instead of native RTSP tracks, which is a combination that is not efficient and can't be used to convert the stream into WebRTC, RTMP or SRT.
In the future, native KLV support will be added to RTSP.

[a comment by an automated account is not shown]
